cloud/shared/bin/lib/terraform.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+import os
+import shutil
+
+from cloud.shared.bin.lib import civiform_mode
+
+
+def perform_apply():
+    '''Generates terraform variable files and runs terraform init and apply.'''
+
+    TERRAFORM_PLAN_OUT_FILE = "terraform_plan"
+    TERRAFORM_TEMPLATE_DIR = os.getenv("TERRAFORM_TEMPLATE_DIR")
+    TF_VARS_FILENAME = os.getenv('TF_VAR_FILENAME')
+
+    terraform_cmd = ['terraform', f'-chdir={TERRAFORM_TEMPLATE_DIR}']
+
+    if civiform_mode.is_dev():
+        subprocess.check_call(
+            terraform_cmd + ['init', '-upgrade', '-reconfigure'])
```

Comment:
Not blocking: Rather than using lists here, I'd consider specifying this as a single string and using `shlex.split`.

Concretely, this one would be something like:

```
subprocess.check_call(shlex.join(f'{terraform_cmd} init -upgrade -reconfigure'))
```

That said, if you think the list approach feels more workable, I'm also fine leaving as-is.